core/sql/common/NAType.cpp

```diff
@@ -670,6 +671,84 @@ short NAType::convertTypeToText(char * text,	   // OUTPUT
 				 displayCaseSpecific);
 }
 
+short NAType::getMyTypeAsHiveText(NAString * outputStr)  // output
+{
+  Lng32		      fs_datatype		= getFSDatatype();
+
+  switch (fs_datatype)
+    {
+    case REC_MIN_F_CHAR_H ... REC_MAX_F_CHAR_H:
+      *outputStr = "string";
+      break;
+
+    case REC_MIN_V_CHAR_H ... REC_MAX_V_CHAR_H:
+      {
+        SQLVarChar * ct = (SQLVarChar*)this;
+        if (ct->wasHiveString())
+          *outputStr = "string";
```

Comment:
yes, that is correct.

Both hive string and varchar datatypes are represented as traf varchar datatypes.
hive varchar has predefined length. hive string length is determined by a session cqd.

The 'wasHiveString' field tells whether it was originally a string datatype. This is
used during showddl command to show the original datatype.

At a later poing, It will also be used to limit the length of data that can be put into a hive column.
For hive varchar, it will be limited by the pre-specified length. For hive string, it will
be limited by the hive length cqd that is set for that session.